### DISCUSSION TITLE
Sword change suggestions: Unique loadout slot, possibly making it a superweapon

### DISCUSSION THREAD

**Discussion by MirceaKitsune** [redeclipse/base#1426] [Ideas and Feature Requests]
For a while I've been thinking of ways to make the sword a more logical and fitting weapon that can better fit in with other weapons. The purpose of this thread is to explain why I think this is necessary and the suggested solution I've thought of over the past weeks. I went with something that shouldn't be radical yet work out very nicely, I think most players can be on board with this suggestion at its core. I've spread it between two main points: Giving the sword its own slot and removing it from the standard loadout, and potentially making it work like a superweapon similar to the rocket... please read the full proposal with an open mind and let me know how you think this would play out and what alternatives if any you believe could be better.

A quick description of what I consider the core issue to be: The sword is a special case within the weapon setup, yet it's not properly categorized as such. What I mean is it's the only melee weapon among guns, included in a setup of projectile based weapons without either looking or functioning as one. As many players like the sword and consider it a staple part of RE, I won't suggest removing or replacing it entirely... however I believe it should be reclassified in its functionality. Ironically my suggestion should make fans of the sword happy, as it would be correctly recognized as a special item as well as made accessible at all times.

## Proposal 1: Using an unique loadout slot

In terms of how the sword is offered, there's both a thematic and practical inconsistency. Thematically I don't think players should have one of their main loadout slots filled by a small and lightweight blade: The sword feels designed be a compact accessory that can be worn separately, you shouldn't have to trade your normal choice of guns for it. It seems right for it to have its own slot like the grenade / mine / rocket; Anyone can get it without having to sacrifice one of their normal guns, be free to wield the sword no matter which two weapons you've picked! Every map already features the sword someplace so this won't be a problem... which brings me to the second issue with the existing configuration.

The energy sword doesn't use any ammo. Not only does this make it even weirder for a gun slot, but it creates an obvious issue: The sword pickup on maps is useless by default and will never spawn in normal circumstances. Ever notice how each map has a blue pad but no item above it? The sword pickup only does anything if you use the Classic mutator to disable loadouts. To me this feels like a broken configuration, a core weapon featured on all maps shouldn't require a mutator to spawn.

Missing pickups can be fixed either by an unique slot or having the sword use ammo, which given it's an **energy** sword would make sense. But since the sword feels like it should be an unique accessory, I'm much in favor of giving it its own slot instead. One outcome is keeping everything else the same: You have the sword forever until you die and need to pick it up again after respawning. I thought of a way to make it even better which brings us to the second part.

## Proposal 2: The sword as a superweapon lasting for a limited time

This would require the first proposal to happen and is an optional addition to it. As I said the sword is a strange weapon among the usual guns and even using it feels so different from everything else, so why not officially make it something special? We only have the rocket as a bonus item: The sword could make for a great complementary superweapon with a few changes.

Its uniqueness would be that once picked up, you only have it for a fixed amount of time as it discharges... 30 seconds is the typical powerup duration in a FPS. Its ammo counter would be a timer that runs out and is reset when you pick up the sword again. This would be accompanied by making the sword more deadly, while the speed boost can be much greater and you may use it to fly at super speeds not possible with normal parkour. In exchange its respawn rate would be decreased so like the rocket it doesn't reappear for a longer amount of time after being picked up.

There's a particular reason for this suggestion: Every classic FPS has an item called Quad Damage, which makes you do enormous amounts of damage for about 30 seconds after taking it. Red Eclipse uses a simple approach of having only weapons as pickups... no health items, no armor, no quad damage. Making the sword a superweapon that lasts for a limited time would be a logical way to fill this gap: You can use it to move super-fast and deal ultra-damage to nearby enemies but have a limited time to do so. It would essentially be Red Eclipse's quad-damage without breaking the "only pickups are weapons" rule.

It's worth noting that nothing should change when using the Medieval mutator: Players would still spawn with just the sword and no other guns, the sword uses no ammo / timer and lasts forever in that mode.

## Conclusion

Like all major changes this should be thought through as to not disrupt the existing configuration too much... at the same time I think some change is needed to have a consistent weapon setup that makes perfect sense, while even gameplay wise I feel this would be a lot more fun. I strongly believe in at least giving the sword an unique loadout slot so anyone can take it and its map pickups are used, also turning it into a superweapon to replace our lack of quad damage feels like a good plan to consider on top. Please let me know what you think, and if anyone has alternative proposals please mention so we can find the ideal way.

**Comment by MirceaKitsune** [comment]
Something I wanted to ask but didn't add to the initial post as to not make it longer: How much of what I'm suggesting can already be done by changing the server settings? It would definitely help to make a config we can exec to test the ideas first! In this case:

- Is there a weapon var to remove the weapon from loadout and give it its own slot? Or is the difference between starting weapons and utility ones hardcoded? Supporting this would be a good idea for any gun so that its purpose may be modded.
- Can you make a weapon's ammo increase or decrease with time? I presume this can be done with a script function at worst, if not maybe support it as a common weapon var? I understand there is a setting to make the sword use ammo and it's just disabled by default, if so adding ammo support to the sword wouldn't even be require code changes.
- Can you make the ammo pickup of the sword have a longer respawn delay? That would be part of making it a superweapon, alongside increased damage and speed boost which I presume are easy to change from the console.

**Comment by qreeves** [comment]
- Hard coded.
```C++
enum
{
    W_CLAW = 0, W_PISTOL, W_OFFSET, // end of unselectable weapon set
    W_SWORD = W_OFFSET, W_SHOTGUN, W_SMG, W_FLAMER, W_PLASMA, W_ZAPPER, W_RIFLE, W_ITEM,
    W_GRENADE = W_ITEM, W_MINE, W_ROCKET, W_ALL, // end of item weapon set
    W_MELEE = W_ALL, W_MAX, // if you add to this at all, check all arrays with W_MAX
    W_LOADOUT = W_ITEM-W_OFFSET,
    W_ITEMS = W_MINE-W_GRENADE+1,
    W_REPLACE = W_GRENADE
};
```
- No, and I wouldn't really want to do it that way for various reasons, especially network sync issues.
- Yes, `/swordspawntime`

**Comment by MirceaKitsune** [comment]
An update worth mentioning: I understand this change will happen when the sword is converted to a spear, DC is also in support of it from on what I remember him mentioning on Discord. The spear will be turned into a powerup weapon like the rocket, limited timer ammo so it only lasts for a fixed amount of time after being picked up. I think that makes a lot of sense, with the sword / spear being an odd case and 6 loadout weapons being plenty enough in my opinion.

**Reply by dc-redeclipse** [reply]
yes should be a time limited pickup like the rocket acting as a short distance alternative to the rocket with likely a similar concept as the rocket.
This is in my opinion the best way to hold balance. Making the Speer as a second permanent weapons like the sword is not useful having two similar weapons.
